docs/testbed/README.testbed.Testbed_v2.hld.md

```diff
@@ -0,0 +1,127 @@
+# Testbed V2 Design
+# High Level Design Document
+
+
+# Table of Contents
+  * [About this Manual](#about-this-manual)
+  * [Requirement Overview](#Requirement-Overview)
+  * [Components](#Components)
+  * [Database Schema](#Database-Schema)
+  * [Implementation Plan](#Implementation-Plan)
+
+
+## About this Manual
+This documentation provides general information about the Testbed V2 feature implementation for `sonic-mgmt`. Testbed V2 aims to support multi-DUTs deployment with dynamic VLAN assignment and inter-DUTs link state propagation.
+
+## Requirement Overview
+* Maintain compatibility with existing testbed.
+* Support testbed/topology with multiple DUTs that have inter-DUTs connections.
+* Support flexible VLAN assignment.
+
+## Components
+* `test user`
+* `connection_db`
+  * Redis databases running on certain test server hosting connection-related metadata.
+* `servercfgd`
+  * running on the same server as `connection_db`
```

Comment:
This is probably out of scope for phase-1, but this is going to get tricky later on if/when we add multiple replicas for the Redis connection DB. Redis _should_ handle the database operations correctly but managing concurrency between servercfgd instances is going to be a challenge.

Reply:
Do we need multiple instances of `servercfgd` for a `Redis` cluster? Only one `servercfgd` running over the master is sufficient, is it correct?

Reply:
My understanding was that the idea behind setting up a redis cluster was so that if the server running the redis DB goes down we can just failover to the other servers. If we only have one servercfgd running on one server, don't we end up back in the same situation?

Reply:
Yes, that's something we need to consider when enabling cluster for `connection_db`. When there are mutliple replicas, `servercfgd` will always runs over the master, there is no need to have `servercfgd` running over slaves since all its functionality is to provision database.

Reply:
That makes sense to me, thanks!